src/iso19111/crs.cpp

```diff
@@ -881,8 +881,14 @@ static bool mustAxisOrderBeSwitchedForVisualizationInternal(
     const std::vector<cs::CoordinateSystemAxisNNPtr> &axisList) {
     const auto &dir0 = axisList[0]->direction();
     const auto &dir1 = axisList[1]->direction();
-    if (&dir0 == &cs::AxisDirection::NORTH &&
-        &dir1 == &cs::AxisDirection::EAST) {
+    if ((&dir0 == &cs::AxisDirection::NORTH &&
+         &dir1 == &cs::AxisDirection::EAST) ||
+        (&dir0 == &cs::AxisDirection::EAST &&
+         &dir1 == &cs::AxisDirection::SOUTH) ||
+        (&dir0 == &cs::AxisDirection::SOUTH &&
+         &dir1 == &cs::AxisDirection::WEST) ||
+        (&dir0 == &cs::AxisDirection::WEST &&
+         &dir1 == &cs::AxisDirection::NORTH)) {
```

Comment:
I'm not convinced that just switching the axis is enough for those 3 coordinate system. We should also presumably transform southing to northing, westing to easting, and make sure the final order is easting, northing

Reply:
That is also changing other problem, that westing-southing is displayed in QGIS with Antartica at the top. In that case moving up in the screen will make the coordinates decrease. Is that normal?

I do not mean that it should not be changed (well, I got used to it in South Africa). But the initial problem in QGIS was the mirroring effect.

Reply:
> In that case moving up in the screen will make the coordinates decrease. Is that normal?

I guess so.

I believe all those coordinate systems should have easting,northing axis order after normalize_for_visualization. The native orders are just conventions to avoid specifying negative values, or other esoteric reasons, but ultimately you want your map with increasing eastings towards the right of the screen, and increasing northings towards the top of the screen

Maybe @thareUSGS can confirm for planetary CRS how maps in a CRS like  IAU_2015:19901 "Mercury (2015) / Ographic" which has northing,westing  should be displayed. I presume this should be transformed as easting,northing ?